### PR TITLE
Updates based on API review feedback

### PR DESCRIPTION
**Pull request by jasonsandlin**

### CODE REVIEW

File: Include/httpClient/httpClient.h [hunk 1]
```diff
@@ -3,12 +3,14 @@
 
 #pragma once
 #include "types.h"
+#include "task.h"
+#include "httpProvider.h"
```
Comment (lubeltra) [inline Include/httpClient/httpClient.h]: I think Cullen was suggesting that this doesn't need to be included in the base header, only people implementing a provider need it

File: Include/httpClient/httpClient.h [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
+#include "mock.h"
```
Comment (lubeltra) [inline Include/httpClient/httpClient.h]: Same for this header, should only be included by people that want to do mocks, where they implement the mocks

File: Include/httpClient/httpClient.h [hunk 3]
```diff
@@ -679,101 +286,109 @@ HCHttpCallRequestGetRequestBodyString(
...
 /// Defaults to true 
 /// </summary>
-/// <param name="call">The handle of the HTTP call</param>
+/// <param name="call">The handle of the HTTP call.  Pass nullptr to set the default for future calls</param>
 /// <param name="retryAllowed">If retry is allowed for this HTTP call</param>
 HC_API void HC_CALLING_CONV
 HCHttpCallRequestSetRetryAllowed(
-    _In_ HC_CALL_HANDLE call,
+    _In_opt_ HC_CALL_HANDLE call,
     _In_ bool retryAllowed
```
Comment (lubeltra) [inline Include/httpClient/httpClient.h]: Bool is not a type in C

File: Include/httpClient/httpClient.h [hunk 4]
```diff
@@ -783,19 +398,8 @@ HCHttpCallRequestGetTimeout(
...
-/// <summary>
-/// Set the response body string of the HTTP call
-/// </summary>
-/// <param name="call">The handle of the HTTP call</param>
-/// <param name="responseString">the response body string of the HTTP call</param>
-HC_API void HC_CALLING_CONV
-HCHttpCallResponseSetResponseString(
-    _In_ HC_CALL_HANDLE call,
-    _In_ PCSTR_T responseString
+    _In_opt_ HC_CALL_HANDLE call,
```
Comment (lubeltra) [inline Include/httpClient/httpClient.h]: Why is this optional? (also applies to the other getters below)

File: Include/httpClient/trace.h [hunk 5]
```diff
@@ -267,15 +267,15 @@ HC_API void HC_CALLING_CONV HCTraceSetClientCallback(HCTraceCallback* callback);
 
 #define HC_PRIVATE_TRACE_AREA_NAME(area) g_trace##area
 
-#if HC_CHAR_T_IS_WIDE
+#if HC_PCSTR_IS_WIDE
```
Comment (lubeltra) [inline Include/httpClient/trace.h]: This should be unnecessary now

File: Samples/Win32Http/main.cpp [hunk 6]
```diff
@@ -60,19 +60,18 @@ DWORD g_numActiveThreads = 0;
...
     {
-        HCTaskGetPendingHandle(),
         g_stopRequestedHandle.get()
     };
 
     bool stop = false;
     while (!stop)
     {
-        DWORD dwResult = WaitForMultipleObjectsEx(2, hEvents, false, INFINITE, false);
+        DWORD dwResult = WaitForMultipleObjectsEx(1, hEvents, false, 20, false);
```
Comment (lubeltra) [inline Samples/Win32Http/main.cpp]: You could use WaitForSingleObject now

File: Source/HTTP/httpcallrequest.cpp [hunk 7]
```diff
@@ -9,8 +9,8 @@ using namespace xbox::httpclient;
 HC_API void HC_CALLING_CONV
 HCHttpCallRequestSetUrl(
     _In_ HC_CALL_HANDLE call,
-    _In_ PCSTR_T method,
-    _In_ PCSTR_T url
+    _In_ PCSTR method,
```
Comment (lubeltra) [inline Source/HTTP/httpcallrequest.cpp]: Why not use an enum for the method?

File: Source/Logger/trace.cpp [hunk 8]
```diff
@@ -15,12 +15,12 @@ namespace
 {
 
 //------------------------------------------------------------------------------
```
Comment (lubeltra) [inline Source/Logger/trace.cpp]: This entire section can be deleted (I'm kind of surprised it even compiles right now)

File: Source/Logger/trace.cpp [hunk 9]
```diff
@@ -165,9 +165,9 @@ void TraceMessageToDebugger(
     std::tm      fmtTime = {};
     localtime_s(&fmtTime, &timeTInSec);
 
-    CHAR_T outputBuffer[BUFFER_SIZE] = {};
+    CHAR outputBuffer[BUFFER_SIZE] = {};
```
Comment (lubeltra) [inline Source/Logger/trace.cpp]: char (lowercase)

Comment (lubeltra) [reply Source/Logger/trace.cpp]: happens in several places

File: Include/httpClient/httpClient.h (hunk 4 above)
Comment (jasonsandlin) [reply Include/httpClient/httpClient.h]: As doc'd in header if null, then it sets default.

File: Source/Logger/trace.cpp (hunk 9 above)
Comment (jasonsandlin) [reply Source/Logger/trace.cpp]: fixed

File: Include/httpClient/trace.h (hunk 5 above)
Comment (jasonsandlin) [reply Include/httpClient/trace.h]: removed

File: Include/httpClient/httpClient.h (hunk 3 above)
Comment (jasonsandlin) [reply Include/httpClient/httpClient.h]: Should be alright.  https://stackoverflow.com/a/1608349/7655781

File: Source/Logger/trace.cpp (hunk 8 above)
Comment (jasonsandlin) [reply Source/Logger/trace.cpp]: Removed

File: Include/httpClient/httpClient.h (hunk 3 above)
Comment (lubeltra) [reply Include/httpClient/httpClient.h]: make sure you include stdbool.h then

I've definitely seen vs barf on it without any includes

File: Include/httpClient/httpClient.h (hunk 4 above)
Comment (lubeltra) [reply Include/httpClient/httpClient.h]: There is a default response string?

I guess that can make sense, seems a bit weird though

File: Source/Logger/trace.cpp (hunk 8 above)
Comment (lubeltra) [reply Source/Logger/trace.cpp]: Doesn't look you have fixed the places these functions were being called

File: Include/httpClient/trace.h [hunk 10]
```diff
@@ -266,12 +266,7 @@ HC_API void HC_CALLING_CONV HCTraceSetClientCallback(HCTraceCallback* callback);
 // DO NOT USE THESE SYMBOLS DIRECTLY
 
 #define HC_PRIVATE_TRACE_AREA_NAME(area) g_trace##area
-
-#if HC_PCSTR_IS_WIDE
-#define HC_FUNCTION __FUNCTIONW__
-#else
 #define HC_FUNCTION __FUNCTION__
```
Comment (lubeltra) [inline Include/httpClient/trace.h]: This is unnecessary, we can just use the standard __FUNCTION__ macro now

File: Include/httpClient/httpClient.h (hunk 4 above)
Comment (jasonsandlin) [reply Include/httpClient/httpClient.h]: Oh right, this one shouldn't be _In_opt_

File: Source/Logger/trace.cpp [hunk 11]
```diff
@@ -14,42 +14,6 @@
...
-
-void OutputDebugStringT(wCHAR const* string)
-{
-    OutputDebugStringW(string);
-}
-
-#else
 
 template<size_t SIZE>
 int stprintf_s(char(&buffer)[SIZE], _Printf_format_string_ char const* format ...)
```
Comment (lubeltra) [inline Source/Logger/trace.cpp]: Now that my crazy overloads are gone, we need to change the code that calls to them

Comment (jasonsandlin) [reply Source/Logger/trace.cpp]: Feel free to fix in a new PR

**Pull request merged**